scylla_operator/conftest.py

```diff
@@ -0,0 +1,102 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+
+from typing import Optional
+import contextlib
+
+import pytest
+
+from scylla_operator.libs.auxiliary import ScyllaOperatorFunctionalClusterTester
+from sdcm.cluster_k8s import ScyllaPodCluster
+
+
+TESTER: Optional[ScyllaOperatorFunctionalClusterTester] = None
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # Populate test result to test function instance
+    outcome = yield
+    rep = outcome.get_result()
+    if rep.passed:
+        item._test_result = ('SUCCESS', None)
+    else:
+        item._test_result = ('FAILED', str(rep.longrepr))
+
+
+@pytest.fixture(autouse=True)
+def harvest_test_results(request, tester):
+    # Pickup test results at the end of the test and submit it to the tester
+
+    def publish_test_result():
+        tester.update_test_status(request.node.nodeid, *request.node._test_result)
+
+    request.addfinalizer(publish_test_result)
+    return None
+
+
+@pytest.fixture(autouse=True, scope='package')
+def tester():
+    tester = ScyllaOperatorFunctionalClusterTester()
+    tester.setUpClass()
+    tester.setUp()
+    tester.healthy_flags = True
+    yield tester
+    with contextlib.suppress(Exception):
+        tester.tearDown()
+    with contextlib.suppress(Exception):
+        tester.tearDownClass()
+
+
+def skip_if_cluster_requirements_not_met(request, tester, db_cluster):
+    require_node_terminate = request.node.get_closest_marker('require_node_terminate')
+    require_mgmt = request.node.get_closest_marker('require_mgmt')
+    if require_node_terminate and require_node_terminate.args:
+        supported_methods = getattr(db_cluster, 'node_terminate_methods', None) or []
+        for terminate_method in require_node_terminate.args:
+            if terminate_method not in supported_methods:
+                pytest.skip(f'cluster {type(db_cluster).__name__} does not support {terminate_method} '
+                            'node termination method')
+    if require_mgmt:
+        if not tester.params.get('use_mgmt'):
+            pytest.skip(f'test require scylla manager to be deployed')
+
+
+@pytest.fixture()
+def db_cluster(tester: ScyllaOperatorFunctionalClusterTester):
+    if tester.healthy_flags:
+        try:
+            if len(tester.db_cluster.nodes) < 3:
+                tester.db_cluster.add_nodes(3 - len(tester.db_cluster.nodes))
+            elif len(tester.db_cluster.nodes) > 3:
+                for node in tester.db_cluster.nodes[3:]:
+                    tester.db_cluster.decommission(node)
+            tester.db_cluster.wait_for_pods_readiness(pods_to_wait=1, total_pods=len(tester.db_cluster.nodes))
+        except:
+            tester.healthy_flags = False
```

Comment:
It looks strange. Why do you depend on exactly 3 nodes and why you make implicit changes to cluster?
I would expect failure when cluster doesn't satisfy our needs.